apps/opik-backend/src/main/java/com/comet/opik/domain/SpanDAO.java

```diff
@@ -444,6 +493,102 @@ public Mono<Void> insert(@NonNull Span span) {
                 .then();
     }
 
+    @Trace(dispatcher = true)
+    public Mono<Void> batchInsert(@NonNull List<Span> spans) {
+        return Mono.from(connectionFactory.create())
+                .flatMapMany(connection -> {
+                    if (spans.isEmpty()) {
+                        return Mono.just(0L);
+                    }
```

Comment:
This isn't possible from the resource as the minimum size is 1. Better validate that spans argument isn't null or empty and fail otherwise. You can use Guava's `Preconditions.checkArgument` method to easily implement it.

This should be an abnormal condition and it's better to fail instead of silently not inserting anything.

Reply:
Yeah, it is more a defensive validation since it's a public method. But I'm changing to use `Preconditions`

Reply:
Done